From: Sam Ravnborg <sam@ravnborg.org>
Date: Sun, 10 Jun 2018 11:19:42 +0200
Subject: [PATCH 603/604] skovarm9cpu: fix DT tree

- Fix pincntrl for display selection
- Drop unused boot mode as key
- Deleted unused gpio-hog entries

Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
---
 arch/arm/dts/skovarm9cpu.dts | 68 +++++++++-----------------------------------
 1 file changed, 13 insertions(+), 55 deletions(-)

diff --git a/arch/arm/dts/skovarm9cpu.dts b/arch/arm/dts/skovarm9cpu.dts
index 7723031a6866..2055c68dfe4a 100644
--- a/arch/arm/dts/skovarm9cpu.dts
+++ b/arch/arm/dts/skovarm9cpu.dts
@@ -87,48 +87,19 @@
 					};
 				};
 
-				inputs {
+				display_type_inputs {
 					pinctrl_disp_type: disp_type-0 {
+						/* Pull-up (HIGH) if Seiko display, otherwise Logic display */
 						atmel,pins =
-							<AT91_PIOD 2 AT91_PERIPH_A AT91_PINCTRL_PULL_UP>;    	      	/* PD2 gpio DISP_TYPE (Seiko or Logic) */
+							<AT91_PIOD 2 AT91_PERIPH_GPIO AT91_PINCTRL_PULL_UP>;
 					};
 					pinctrl_logic_type: logic_type-0 {
+						/* Pull-down (LOW) if l6whrt display, otherwise l2rt display */
 						atmel,pins =
-							<AT91_PIOB 8 AT91_PERIPH_A AT91_PINCTRL_PULL_UP>;                /* PB8 gpio LOGIC_TYPE (l2 or l6) */
+							<AT91_PIOB 8 AT91_PERIPH_GPIO AT91_PINCTRL_PULL_UP>;
 					};
 				};
-				/* clk appear not to be ready at first probe.
-				 * so this hangs barebox if enabled
-				 * pioD: gpio@fffff800 {
-				 *	DISP_TYPE {
-				 *		gpio-hog;
-				 *		input;
-				 *		gpios = <&pioD 2>;
-				 *		line-name = "DISP_TYPE";
-				 *	};
-				 * };
-				 *
-				 */
-
-				pioB: gpio@fffff400 {
-					/* Enable the 50MHz oscillator for Ethernet PHY */
-					/*
-					 * Will cause null pointer exception in the kernel
-					 *phy_50mhz {
-					 *	gpio-hog;
-					 *	gpios = <27 GPIO_ACTIVE_LOW>;
-					 *	output-high;
-					 *	line-name = "PHY 50 MHz oscillator";
-					 *};
-					 *
-					 *LOGIC_TYPE {
-					 *	gpio-hog;
-					 *	input;
-					 *	gpios = <&pioB 8>;
-					 *	line-name = "LOGIC_TYPE";
-					 *};
-					 */
-				};
+
 				fb {
 					pinctrl_board_fb: fb-0 {
 						atmel,pins =
@@ -209,11 +180,15 @@
 				atmel,guard-time = <1>;
 				atmel,lcd-wiring-mode = "BRG";
 				pinctrl-names = "default";
-				pinctrl-0 = <&pinctrl_board_fb>;
+				pinctrl-0 = <
+					&pinctrl_board_fb
+					&pinctrl_disp_type
+					&pinctrl_logic_type>;
+
 				atmel,power-control-gpio = <&pioA 30 GPIO_ACTIVE_HIGH>;
 
 				display-timings {
-					native-mode = <&l6rt>;
+					native-mode = <&l2rt>;
 					l2rt: l2rt {
 						clock-frequency = <33000000>;
 						hactive = <800>;
@@ -229,7 +204,7 @@
 						vsync-active = <0>;
 						de-active = <1>;
 					};
-					l6rt: l6rt {
+					l6whrt: l6whrt {
 						clock-frequency = <33000000>;
 						hactive = <800>;
 						vactive = <480>;
@@ -324,23 +299,6 @@
 		};
 	};
 
-	gpio_keys {
-		compatible = "gpio-keys";
-
-		boot_mode_select {
-			label = "boot mode";
-			gpios = <&pioB 3 GPIO_ACTIVE_LOW>;
-			linux,code = <272>;
-		};
-	};
-
-	inputs {
-		compatible = "atmel,at91sam9x5-pinctrl, atmel,at91rm9200-pinctrl", "simple-bus";
-		pinctrl-names = "default";
-		pinctrl-0 = <
-			&pinctrl_disp_type
-			&pinctrl_logic_type>;
-	};
 
 	i2c-gpio-0 {
 		status = "okay";
